src/listeners/erldns_proto_udp.erl

```diff
@@ -88,16 +89,19 @@ handle_decoded(Socket, IpAddr, Port, DecodedMessage, TS0) ->
     ?LOG_DEBUG(#{what => tcp_request, request => DecodedMessage, response => Response}),
     measure_time(DecodedMessage, EncodedResponse, TS0).
 
--spec max_payload_size(dns:message()) -> pos_integer().
+-spec max_payload_size(dns:message()) -> dns:message().
 max_payload_size(Message) ->
```

Comment:
Have you considered renaming the function to be a little more descriptive of it's intent e.g. `normalize_edns_max_payload_size`?

Reply:
Good point, doing in a sec